src/components/schema-editor/schema-editor.component.tsx

```diff
@@ -2,49 +2,132 @@ import React from 'react';
+import { Toggletip, ToggletipContent, ToggletipButton, ToggletipActions, Link } from '@carbon/react';
+import { WarningAlt } from '@carbon/react/icons';
+import type { Schema } from '../../types';
 import styles from './schema-editor.scss';
 
+interface markerProps {
+  startCol: number;
+  startRow: number;
+  endCol: number;
+  endRow: number;
+  className: string;
+  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents, @typescript-eslint/no-explicit-any
+  type: string | any;
+  text: string;
+}
 interface SchemaEditorProps {
   isLoading: boolean;
-  invalidJsonErrorMessage: string;
   onSchemaChange: (stringifiedSchema: string) => void;
   stringifiedSchema: string;
+  errors: Array<markerProps>;
+  setErrors: (errors: Array<markerProps>) => void;
 }
 
-const SchemaEditor: React.FC<SchemaEditorProps> = ({ invalidJsonErrorMessage, onSchemaChange, stringifiedSchema }) => {
-  const { t } = useTranslation();
+const SchemaEditor: React.FC<SchemaEditorProps> = ({ onSchemaChange, stringifiedSchema, errors, setErrors }) => {
+  const fetchAndParseSchema = async () => {
+    const response = await fetch('https://json.openmrs.org/form.schema.json');
+    const schema: Schema = await response.json();
+    return schema;
+  };
 
-  return (
-    <>
-      {invalidJsonErrorMessage ? (
-        <div className={styles.errorContainer}>
-          <p className={styles.heading}>{t('schemaError', "There's an error in your schema.")}</p>
-          <p>{invalidJsonErrorMessage}</p>
-        </div>
-      ) : null}
+  const validateJSON = (content: string, schema) => {
+    const ajv = new Ajv({ allErrors: true, jsPropertySyntax: true });
+    const validate = ajv.compile(schema);
+    const parsedContent = JSON.parse(content);
+    const isValid = validate(parsedContent);
+
+    if (!isValid) {
+      const jsonLines = content.split('\n');
+      const errorMarkers = validate.errors.map((error) => {
+        let lineNumber = -1;
+        const instancePath = error.instancePath.replace(/^\./, '');
+        for (let i = 0; i < jsonLines.length; i++) {
+          if (jsonLines[i].includes(instancePath)) {
+            lineNumber = i;
+            break;
+          }
+        }
+        const message =
+          error.keyword === 'type'
+            ? `Invalid Type: ${error.instancePath.substring(1)} ${error.message}`
+            : `${capitalize(Object.keys(error.params)[0])}: ${
+                error.message.charAt(0).toUpperCase() + error.message.slice(1)
+              }`;
+
+        return {
+          startRow: lineNumber,
+          startCol: 0,
+          endRow: lineNumber,
+          endCol: 1,
+          className: 'error',
+          text: message,
+          type: 'error',
+        };
+      });
+      setErrors(errorMarkers);
+    } else {
+      setErrors([]);
+    }
+  };
 
+  const handleChange = (newValue: string) => {
+    onSchemaChange(newValue);
+    fetchAndParseSchema()
+      .then((schema) => {
```

Comment:
In relation to the comment above, I think we should have this network request as a hook at the top of the component, making this fetch request on every change seems like an expensive process, also might be more efficient making this request using SWR.

Reply:
Sure.